x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/apikey/RestInvalidateApiKeyAction.java

```diff
@@ -78,4 +76,43 @@ public RestResponse buildResponse(InvalidateApiKeyResponse invalidateResp,
     public String getName() {
         return "xpack_security_invalidate_api_key";
     }
+
+    private ConstructingObjectParser<InvalidateApiKeyRequest, Void> getObjectParser(RestRequest request) {
+        if (request.getRestApiVersion() == RestApiVersion.V_7) {
+            final ConstructingObjectParser<InvalidateApiKeyRequest, Void> objectParser = new ConstructingObjectParser<>(
```

Comment:
Is there a reason this isn't a `static final` field like the non-compat parser?
I'm not particularly fussed either way, it just surprised me.